src/main/flight/flight_health.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <math.h>
+
+#include "platform.h"
+
+#include "build/debug.h"
+
+#include "common/axis.h"
+#include "common/utils.h"
+#include "common/time.h"
+
+#include "fc/core.h"
+#include "fc/runtime_config.h"
+#include "fc/rc_controls.h"
+
+#include "flight/mixer.h"
+#include "flight/pid.h"
+
+#include "sensors/gyro.h"
+
+#include "config/feature.h"
+#include "config/config.h"
+
+#include "flight_health.h"
+
+#define THRUST_IMBALANCE_GYRO_RATE_MAX_DPS 200 // degrees per second
+#define THRUST_IMBALANCE_THROTTLE_MIN_PCT 10 // percent
+#define THRUST_IMBALANCE_BLANKING_TIME_US 500000 // 0.5 sec
+#define THRUST_IMBALANCE_CONFIG_TO_WORKING_DOMAIN 1e5f
```

Comment:
```suggestion
#define THRUST_IMBALANCE_GYRO_RATE_MAX_DPS 200 // disable detection with some gyro axis above this threshold
#define THRUST_IMBALANCE_THROTTLE_MIN_PCT 10 //  disable detection when throttle is below this threshold
#define THRUST_IMBALANCE_BLANKING_TIME_US 500000 // 0.5 sec; delay enabling of detection
#define THRUST_IMBALANCE_CONFIG_TO_WORKING_DOMAIN 1e5f ; see code
```

`threshold` should be computed as (`thrust_imbalance_threshold` * SOME_SCALING_FACTOR) ^ 3. It will allow much greater control over threshold range and user will specify value that is directly related to PID I term.
(mathematically, geometric mean of I-term should be used. But third power is faster than cube root)